src/components/SelectionList/ReportListItem.tsx

```diff
@@ -0,0 +1,168 @@
+                            participantTo={participantTo}
+                            buttonText={translate('common.view')}
+                            onButtonPress={handleOnButtonPress}
+                        />
+                    )}
+                    <View style={[styles.flex1, styles.flexRow, styles.alignItemsCenter, isLargeScreenWidth && styles.mr4]}>
+                        {/** marginRight added here to move the action button by the type column distance */}
+                        <View style={[styles.flexRow, styles.flex1, styles.alignItemsCenter, styles.justifyContentBetween, isLargeScreenWidth && {marginRight: TYPE_COLUMN_WIDTH}]}>
+                            <View style={[styles.flexRow, styles.alignItemsCenter, styles.flex2]}>
+                                {canSelectMultiple && (
+                                    <ListItemCheckbox
+                                        accessibilityLabel={item.text}
+                                        isDisabled={Boolean(isDisabled)}
+                                        isDisabledCheckbox={Boolean(item.isDisabledCheckbox)}
+                                        isSelected={Boolean(item.isSelected)}
+                                        onPress={() => {}}
+                                    />
+                                )}
+
+                                <View style={[styles.flexShrink1, isLargeScreenWidth && styles.ph4]}>
+                                    <Text style={[styles.reportListItemTitle]}>{reportItem?.reportName}</Text>
+                                    <Text style={[styles.textMicroSupporting]}>{`${reportItem.transactions.length} grouped expenses`}</Text>
+                                </View>
+                            </View>
+                            <View style={[styles.flexRow, styles.flex1, styles.justifyContentEnd]}>{totalCell}</View>
+                        </View>
+                        {isLargeScreenWidth && <View style={[StyleUtils.getSearchTableColumnStyles(CONST.SEARCH_TABLE_COLUMNS.ACTION)]}>{actionCell}</View>}
+                    </View>
+                    <View style={[styles.mt3, styles.reportListItemSeparator]} />
+                    {reportItem.transactions.map((transaction) => (
+                        <TransactionListItemRow
+                            item={transaction}
+                            showTooltip={showTooltip}
+                            isDisabled={Boolean(isDisabled)}
+                            canSelectMultiple={Boolean(canSelectMultiple)}
+                            onButtonPress={() => {
+                                openReportInRHP(transaction);
+                            }}
+                            showItemHeaderOnNarrowLayout={false}
+                            containerStyle={styles.mt3}
+                        />
+                    ))}
+                </View>
+            </BaseListItem>
+        );
+    }
+
+    return null;
```

Comment:
please reverse the conditions. First check if we don't want to display anything and `return null`
Then return the actual item.
This order looks very weird - we should prefer early returns.

Also can you tell me why this case even exists? that is why this component might not return anything?